src/view/show_mode/editor/show_ui_widgets/cue_control.py

```diff
@@ -35,6 +49,9 @@ def __init__(self, fid: str, parent: UIPage, filter_model: Filter | None, config
         self._input_dialog: QInputDialog | None = None
         self._dialog_widget: QWidget | None = None
 
+    def __exit__(self, exc_type, exc_value, traceback):
+        self._filter.scene.board_configuration.broadcaster.update_filter_parameter.disconnect(self._cue_state.update)
+
```

Comment:
i just loaded a new showfile, but the old callback of update does not get deleted. Where we could put it?
should i change it, that the cue_state belongs to (a new class?) CueFilter inheriting from Filter, and add a method to Filter: remove(), which can call the scene before getting deleted in board_configuration:54 _delete_scene(self, scene: Scene)
Until now there are no classes for each filter, or did i miss them? They are all just Filter with different attributes...?
Or other approach: What happens when switching scenes? The UI widgets are getting hidden? But when loading a showfile, or deleting a scene, what happens to the UI widgets?

Reply:
First, you are correct, there are no filter classes at the moment. This, however, is subject to change. One might create a class inheriting from filter if that's desired. (It would probably make sense but it would be a task for a later time).
Second, as of now, the editor does not really support loading a show file once there is already one loaded or content placed. Ideally one would have some kind of unloading logic prior to loading a new show file but this is also a remaining todo.